core/src/util.rs

```diff
@@ -67,6 +70,22 @@ pub trait AsyncSleeping: Send + Sync {
 pub struct AsyncSleep;
 impl AsyncSleeping for AsyncSleep {}
 
+#[cfg_attr(test, mockall::automock)]
+pub trait Now: Send + Sync {
```

Comment:
I see how this can be useful for testing things that rely on specific times.

Reply:
Agree. In this case maybe a Fake implementation with a `set_time` method rather than a mock (that requires slightly more involved expectation setup) could be convenient, but we can add this when we think it's useful.